### ISSUE TITLE
reportUnknownArgumentType in 1.1.286

### ISSUE THREAD

**Issue by mehdigmira** [microsoft/pyright#4386]
reportUnknownArgumentType error is reported in 1.1.286 but not 1.1.285. 

```python
# pyright: strict


my_dict: dict[tuple[int, int], int] = {}
items: set[int] = set()

for _ in range(10):
    x = 1

for _ in range(10):
    x = 1
    a = my_dict.get((1, 2))
    if a:
        item = my_dict.get((x, x))
        if item and item not in items:
            items.add(item)
```

Pyright reports `error: Argument type is partially unknown
    Argument corresponds to parameter "__key" in function "get"
    Argument type is "tuple[Unknown, Unknown]" (reportUnknownArgumentType)`, while using reveal_type((x,x)) outputs `information: Type of "(x, x)" is "tuple[Literal[1], Literal[1]]"`, which means types are inferred correctly.

Sorry if the code above makes no sense, but that's the only way i was able to reproduce this with a minimal example.

**Comment by erictraut** [comment]
Thanks for the great bug report. Your code sample made it easy to find and fix the bug.

This was a long-standing bug that was unmasked by an improvement I made in 1.1.286. It will be fixed in the next version.

**Comment by rafrafek** [comment]
Is it related to my issue? The code below:

```python3
import pytest

@pytest.fixture(params=[None, 1, 2, 3])
def my_fixture(request: pytest.FixtureRequest):
    assert hasattr(request, "param") and isinstance(request.param, (type(None), int))
    with open("/tmp/foo") as file:
        yield file.read(request.param)
```

gives me error in two places for `request.param`:

<img width="743" alt="image" src="https://user-images.githubusercontent.com/23004737/210371923-68867c01-8fc3-4f1e-b7d6-dee76327a953.png">

Error says:

> param: int | None
> Type of "param" is unknown Pylance [reportUnknownMemberType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportUnknownMemberType)
> Cannot access member "param" for type "FixtureRequest"
>   Member "param" is unknown Pylance [reportGeneralTypeIssues](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportGeneralTypeIssues)

<img width="522" alt="image" src="https://user-images.githubusercontent.com/23004737/210372292-f31f3820-532c-4ac6-8547-16b14fa95bc8.png">

**Comment by erictraut** [comment]
@rafrafek, no, this isn't the same problem. I'm not able to repro your issue with pyright 1.1.286. You're likely using an out-of-date version of pytest. Please update to the latest version. If you're still seeing the problem, please file a new bug report with repro details.

**Comment by rafrafek** [comment]
@erictraut, yes, you're right! Bumping pytest from 6.2.5 to 7.2.0 helped. Thanks!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.287, which I just published. It will also be included in a future release of pylance.
